src/io/mod.rs

```diff
@@ -171,8 +171,16 @@ impl<D: DeserializeOwned> DelimFileReader<D> {
             .has_headers(true)
             .quoting(quote)
             .from_reader(reader);
-        assert!(csv_reader.has_headers(), "Expected input file to have a header row");
+
+        // NB: csv_reader.has_header() does not actually check for existence of a header, but only
+        // checks that the reader is configured to read a header.
+
+        // If the header is not empty (try to parse it)
         let header = csv_reader.headers().map_err(FgError::ConversionError)?.to_owned();
+        if !header.is_empty() {
```

Comment:
Empty files return empty headers, so we validate headers for only non-empty TSVs.